openfl-workspace/tf_3dunet_brats/src/dataloader.py

```diff
@@ -0,0 +1,284 @@
+            elif channel == 3:
+                imgfile = self.filenames[idx][1].replace('_flair', '_t2')
+
+            img_temp = np.array(nib.load(imgfile).dataobj)
+
+            img_temp = np.rot90(img_temp)
+
+            # Normalize
+            img_temp = self.z_normalize_img(img_temp)
+
+            img[..., channel] = img_temp
+
+        msk = np.rot90(np.array(nib.load(mskfile).dataobj))
+        msk = np.expand_dims(msk, -1)
+
+        # labels: {
+        #      0: background,
+        #      1: edema,
+        #      2: non-enhancing tumor,
+        #      3: enhancing tumour}
+
+        # Combine all masks but background
+        if self.num_classes == 1:
+            msk[msk > 0] = 1.0
+        else:
+            msk_temp = np.zeros(list(msk.shape) + [self.num_classes])
+            for channel in range(self.num_classes):
+                msk_temp[msk == channel, channel] = 1.0
+            msk = msk_temp
+
+        # Crop
+        img, msk = self.crop(img, msk, randomize)
+
+        # Randomly rotate
+        if randomize:
+            img, msk = self.augment_data(img, msk)
+
+        return img, msk
+
+    def get_input_shape(self):
+        """Get the shape of the input."""
+        return self.crop_dim
+
+    def plot_images(self, ds, slice_num=90):
+        """Plot images from dataset."""
+        import matplotlib.pyplot as plt
+
+        plt.figure(figsize=(20, 20))
+
+        num_cols = 2
+
+        msk_channel = 0
+        img_channel = 0
+
+        for img, msk in ds.take(1):
+            bs = img.shape[0]
+
+            for idx in range(bs):
+                plt.subplot(bs, num_cols, idx * num_cols + 1)
+                plt.imshow(img[idx, :, :, slice_num, img_channel], cmap='bone')
+                plt.title('MRI', fontsize=18)
+                plt.subplot(bs, num_cols, idx * num_cols + 2)
+                plt.imshow(msk[idx, :, :, slice_num, msk_channel], cmap='bone')
+                plt.title('Tumor', fontsize=18)
+
+        plt.show()
+
+        print(f'Mean pixel value of image = {np.mean(img[0, :, :, :, 0])}')
+
+    def display_train_images(self, slice_num=90):
+        """Plot some training images."""
+        self.plot_images(self.ds_train, slice_num)
+
+    def display_validation_images(self, slice_num=90):
+        """Plot some validation images."""
+        self.plot_images(self.ds_val, slice_num)
+
+    def display_test_images(self, slice_num=90):
+        """Plot some test images."""
+        self.plot_images(self.ds_test, slice_num)
+
+    def get_train(self):
+        """Return train dataset."""
+        return self.ds_train
+
+    def get_test(self):
+        """Return test dataset."""
+        return self.ds_test
+
+    def get_validate(self):
+        """Return validation dataset."""
+        return self.ds_val
+
+    def get_dataset(self):
+        """Create a TensorFlow data loader."""
+        self.num_train = int(self.numfiles * self.train_test_split)
```

Comment:
```
AttributeError: 'DatasetGenerator' object has no attribute 'numfiles'
```

Reply:
I guess the flake check missed it. Had to change self.numFiles to self.numfiles. I think this should fix.